pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedPrivateMethodRule.java

```diff
@@ -103,6 +104,25 @@ public Object visit(ASTCompilationUnit file, Object param) {
                         consideredNames.remove(methodName); // clear this name
                     }
                 }
+
+                // TODO workaround for missing method symbol resolution
```

Comment:
did you try to solve the underlaying resolution issue? what issue did you run into doing so?

Reply:
No, I didn't investigate that. In the debugger I saw, that the symbol we got here was net.sourceforge.pmd.lang.java.types.UnresolvedMethodSig.UnresolvedMethodSym

Reply:
I've been playing with this a little…

Either specifying the parameter type as `(String clientId) -> this.notCachedGetAllDps(clientId))` or using a method reference such as `this::notCachedGetAllDps` will not produce a FP.

When using `clientId -> this.notCachedGetAllDps(clientId))` it will infer `clientId` to be a `java.lang.Object`, and then test applicability of the method, which will fail, as `java.lang.Object` can't be used for `java.lang.String`.

The issue seems to be on how infered types for generics are propagated down to the `LambdaExpression`s `LambdaMirrorImpl`